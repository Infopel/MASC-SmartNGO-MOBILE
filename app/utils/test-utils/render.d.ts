export type RenderOptions = {
  wrapper?: React.ComponentType<any>
  createNodeMock?: (element: React.ReactElement) => any
}
export type AllTheProvidersProps = { children: React.ReactNode }
