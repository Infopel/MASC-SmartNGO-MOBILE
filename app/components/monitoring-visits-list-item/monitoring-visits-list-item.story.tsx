import * as React from "react"
import { storiesOf } from "@storybook/react-native"
import { StoryScreen, Story, UseCase } from "../../../storybook/views"
import { MonitoringVisitsListItem } from "./monitoring-visits-list-item"

storiesOf("MonitoringVisitsListItem", module)
  .addDecorator((fn) => <StoryScreen>{fn()}</StoryScreen>)
  .add("Style Presets", () => (
    <Story>
      <UseCase text="Primary" usage="The primary.">
        <MonitoringVisitsListItem m='2' />
      </UseCase>
    </Story>
  ))
