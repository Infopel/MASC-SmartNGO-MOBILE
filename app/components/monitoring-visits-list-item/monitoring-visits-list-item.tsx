import * as React from "react"
import { IBoxProps, Box, HStack } from "native-base"
import { Text } from "components/text/text"
import { MonitoringVisit } from "storage/models/monitoring-visit"
import { DateUtils } from "utils/DateUitls"
import { Icon } from "components/icon/icon"

export interface MonitoringVisitsListItemProps extends IBoxProps {
  item: MonitoringVisit
}

/**
 * Describe your component here
 */
export function MonitoringVisitsListItem({ item: { date, motive, type }, ...rest }: MonitoringVisitsListItemProps) {

  return (
    <Box {...rest} m='4'>
      <Text >{type}</Text>
      <Text >{motive}</Text>
      <HStack alignItems={'center'} space='2' py='2'>
        <Icon name='calendar' size='md'  type="material-community" />
        <Text>{DateUtils.prettyDate(date)}</Text>
      </HStack>
    </Box>
  )
}
