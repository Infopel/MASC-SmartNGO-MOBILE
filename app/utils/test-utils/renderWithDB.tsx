jest.mock("storage/adapter.ts")
import DatabaseProvider from "@nozbe/watermelondb/DatabaseProvider"
import { render } from "@testing-library/react-native"
import { default as React } from "react"
import { database } from "storage/database"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProvidersWithDB({ children }: AllTheProvidersProps) {
    return (
        <DatabaseProvider database={database}>
            <ThemeProvider>{children}</ThemeProvider>
        </DatabaseProvider>
    )
}

export function renderWithDB<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
    return render(ui, { wrapper: AllTheProvidersWithDB, ...options })
}


export class DiagnosticError extends Error {
    name: string = "Diagnostic error"
  }
  