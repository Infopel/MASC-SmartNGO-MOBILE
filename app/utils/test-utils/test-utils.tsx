import { render } from "@testing-library/react-native"
import React from "react"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProviders({ children }: AllTheProvidersProps) {
  return (
    <ThemeProvider>{children}</ThemeProvider>
  )
}

function customRender<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

// re-export everything
export * from "@testing-library/react-native"
// override render method
export { customRender as render }

