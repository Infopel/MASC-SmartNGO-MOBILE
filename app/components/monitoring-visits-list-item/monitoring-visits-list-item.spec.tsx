import { fireEvent, render } from "utils/test-utils/test-utils"
import * as React from "react"
import { MonitoringVisitsListItem } from "./monitoring-visits-list-item"



describe("<MonitoringVisitsListItem>",()=>{
  it("render", async function () {
    const component = <MonitoringVisitsListItem></MonitoringVisitsListItem>
  
    expect(render(component).toJSON()).toMatchSnapshot()
  })
})
