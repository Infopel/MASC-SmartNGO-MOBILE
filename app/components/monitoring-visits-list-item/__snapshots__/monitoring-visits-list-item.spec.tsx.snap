// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MonitoringVisitsListItem> render 1`] = `
<View
  dataSet={Object {}}
  style={
    Object {
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "marginTop": 16,
    }
  }
>
  <Text
    dataSet={Object {}}
    style={
      Object {
        "backgroundColor": undefined,
        "color": "#fb923c",
        "fontFamily": undefined,
        "fontSize": 14,
        "fontStyle": "normal",
        "fontWeight": "400",
        "letterSpacing": 0,
        "lineHeight": 21,
        "textDecorationLine": undefined,
      }
    }
  >
    Hello
  </Text>
</View>
`;
